fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,822 @@
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(kwname, "a boolean")
+                )
+
+        if self.callbacks and not callable(self.callbacks):
+            if not isinstance(self.callbacks, list):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        "callbacks", "a callable or list of callables"
+                    )
+                )
+            else:
+                for cb in self.callbacks:
+                    if not callable(cb):
+                        raise ValueError(
+                            _KWARG_ERROR_MESSAGE.format(
+                                "callbacks", "a callable or list of callables"
+                            )
+                        )
+
+        # NOTE: inferring distribution type should happen before transforming
+        read_kw = (
+            lambda data, kw_or_func: _get_type(data, kw_or_func)
+            if isinstance(kw_or_func, str)
+            else kw_or_func
+        )
+
+        self.predictor_loss_ = read_kw(Y, self.predictor_loss)
+        self.adversary_loss_ = read_kw(Z, self.adversary_loss)
+        self.predictor_function_ = read_kw(Y, self.predictor_function)
+
+        kws = ["binary", "category", "continuous"]
+        for kw, kwname in (
+            (self.predictor_loss_, "predictor_loss"),
+            (self.adversary_loss_, "adversary_loss"),
+            (self.predictor_function_, "predictor_function"),
+        ):
+            # Some keywords can also be a callable instead of a str.
+            if not ((isinstance(kw, str) and kw in kws) or callable(kw)):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        kwname,
+                        (
+                            "'auto', 'binary', 'category', 'continuous', "
+                            + "or a callable",
+                        ),
+                    )
+                )
+
+        for kw, kwname in (
+            (self.y_transform, "y_transform"),
+            (self.z_transform, "z_transform"),
+        ):
+            if not (
+                issubclass(type(kw), TransformerMixin)
+                and hasattr(kw, "fit")
+                and hasattr(kw, "transform")
+            ):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        kwname,
+                        "a sklearn Transformer (subclass TransformerMixin)",
+                    )
+                )
+
+        self.y_transform_ = self.y_transform.fit(Y)
+        self.z_transform_ = self.z_transform.fit(Z)
+
+        if self.cuda and not isinstance(self.cuda, str):
+            raise ValueError(
+                _KWARG_ERROR_MESSAGE.format(
+                    "cuda",
+                    'None, or the device to use (such as "cuda:0")',
+                )
+            )
+
+        self.random_state_ = check_random_state(self.random_state)
```

Comment:
do we need to store this in the instance? It could be used just where it's needed, can't it?

Reply:
Perhaps I interpreted this in the wrong way, but I was following this convention https://scikit-learn.org/stable/developers/develop.html#random-numbers